### PR TITLE
Fix effective principal quantum number `nEff` in ADK

### PR DESCRIPTION
**Pull request by n01r**
Fixes the calculation of the effective principal quantum number `nEff` for
the implemented ADK ionization model. The quantity `nEff` arises from scaling
an ionic potential of an arbitrary configuration and charge state Z to the
hydrogenic Bohr model. In the Bohr model:

    n = Z / sqrt(2 * E_n)

where E_n denotes the energy of the n-th level/shell.
In order to extrapolate that to arbitrary ions one has to do the following
replacements:

    Z   (nuclear charge)    --> Z_i (ionic residual charge)
    E_n (n-th shell energy) --> E_i (ionization energy of the state)

Here atomic residual charge means charge AFTER ionization since this is the
charge that the *electron-to-be-ionized* 'feels'.

**Previously it was wrongly implemented since the bare nuclear charge was used to calculate the effective principal quantum number**

In the following comparison one can see the ionization rates for the transition from C II to C III (meaning 1+ to 2+). 
comparison_carbon_ii
While previously (orange curve *ADK C H-like*) with the old `nEff` values this carbon state was ionized earlier than hydrogen that is not the case anymore (green curve *ADK C (fixed)*).

| state | ionization energy (eV) |
| - | - |
| H | 13.6 |
| C I | 11.26 |
| C II |  24.36 |

While in the hydrogen case the effective charge is just 1 for C II it should be either 2 (*proton number minus number of bound electrons in ionized state*) or 3.136 when accounting for shielding of the orbitals via SCF methods   
(see: Clementi, E.; Raimondi, D. L. (1963). "*Atomic Screening Constants from SCF Functions*". J. Chem. Phys. 38 (11): 2686–2689.)

**This affects all previous simulations done with the ADK ionization model.**
Especially low charge states are affected, as can be seen in the example of values of `nEff` for carbon ions.

| state | C I | C II | C III | C IV | C V | C VI |
| - | - | - | - | - | - | - |
| `nEff` (fixed) | 1.099 | 1.494 | 1.599 | 1.837 | 0.931 |1.000 |
| `nEff` (old) | 6.594 | 4.483 | 3.197 | 2.755 | 1.117 | 1.000 |

That previously led to an overestimation of ionization for low charge states.

### CODE REVIEW

**Comment by ax3l** [comment]
offline discussion: the H-like models will be removed as they only made sense for the last bound e- and are error prune.

The Zeff model in BSI (and an added follow-up for ADK) are interesting for users to tune their rates but are not necessarily better, as we can e.g. see for the BSI/ADK thresholds where a 24eV C2+ e- is estimated with the same threshold as a 13.6 eV Hydrogen e-, which is obviously weird and caused by the common approximation to increase Zeff only per closed shell. Luckily, we allow the user to change that at will, but we need to document that Zeff does not necessarily mean its "better" as in the example here.

for this PR, we will add a description for the sphinx docs with the models and the image above (without the two orange lines) and a follow-up to add ADK with Zeff, although it might be as bad as BSI with Zeff but at least its configurable.

File: docs/source/models/field_ionization_effective_potentials.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    plt.ylabel("$V_\mathrm{eff}$ [AU = Rydberg]",size=fontsize)
+    plt.xlabel("$x$ [AU = Bohr radii]",size=fontsize)
+
+    ax = plt.gca()
+    ax.tick_params(size=fontsize,labelsize=fontsize)
+
+    plt.draw()
+    plt.show()
+
+    plt.savefig("field_ionization_effective_potentials.svg")
```
Comment (n01r) [inline docs/source/models/field_ionization_effective_potentials.py]: missing EOF newline

File: docs/source/models/field_ionization.rst [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        E. Clementi and D. Raimondi.
+        *Atomic Screening Constant from SCF Functions*,
+        The Journal of Chemical Physics 38, 2686-2689 (1963)
+        https://dx.doi.org/10.1063/1.1733573
+
+.. [ClementiRaimondi1967]
+        E. Clementi and D. Raimondi.
+        *Atomic Screening Constant from SCF Functions. II. Atoms with 37 to 86 Electrons*,
+        The Journal of Chemical Physics 47, 1300-1307 (1967)
+        https://dx.doi.org/10.1063/1.1712084
```
Comment (n01r) [inline docs/source/models/field_ionization.rst]: missing EOF newline

File: src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp [hunk 3]
```diff
@@ -79,21 +79,25 @@ namespace ionization
                 /* electric field in atomic units - only absolute value */
                 float_X eInAU = math::abs(eField) / ATOMIC_UNIT_EFIELD;
 
+                /* the charge that attracts the electron that is to be ionized:
+                 * equals `protonNumber - #allInnerElectrons`
+                 */
+                float_X effectiveCharge = chargeState + float_X(1.0);
```
Comment (n01r) [inline src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp]: `float_X const` 

as well as above where:
> const float_X protonNumber = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
> float_X chargeState = attribute::getChargeState(parentIon);

File: src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp [hunk 4]
```diff
@@ -79,21 +79,25 @@ namespace ionization
...
                 /* effective principal quantum number (unitless) */
-                float_X nEff = protonNumber / math::sqrt(float_X(2.0) * iEnergy );
+                float_X nEff = effectiveCharge / math::sqrt(float_X(2.0) * iEnergy );
                 /* nameless variable for convenience dFromADK*/
-                float_X dBase = float_X(4.0) * util::cube(protonNumber) / (eInAU * util::quad(nEff)) ;
+                float_X dBase = float_X(4.0) * util::cube(effectiveCharge) / (eInAU * util::quad(nEff)) ;
                 float_X dFromADK = math::pow(dBase,nEff);
```
Comment (n01r) [inline src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp]: `float_X const`

File: src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp [hunk 5]
```diff
@@ -79,21 +79,25 @@ namespace ionization
...
+                /* the charge that attracts the electron that is to be ionized:
+                 * equals `protonNumber - #allInnerElectrons`
+                 */
+                float_X effectiveCharge = chargeState + float_X(1.0);
                 /* effective principal quantum number (unitless) */
-                float_X nEff = protonNumber / math::sqrt(float_X(2.0) * iEnergy );
+                float_X nEff = effectiveCharge / math::sqrt(float_X(2.0) * iEnergy );
                 /* nameless variable for convenience dFromADK*/
-                float_X dBase = float_X(4.0) * util::cube(protonNumber) / (eInAU * util::quad(nEff)) ;
+                float_X dBase = float_X(4.0) * util::cube(effectiveCharge) / (eInAU * util::quad(nEff)) ;
```
Comment (n01r) [inline src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp]: `float_X const`

File: src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp [hunk 6]
```diff
@@ -79,21 +79,25 @@ namespace ionization
                 /* electric field in atomic units - only absolute value */
                 float_X eInAU = math::abs(eField) / ATOMIC_UNIT_EFIELD;
 
+                /* the charge that attracts the electron that is to be ionized:
+                 * equals `protonNumber - #allInnerElectrons`
+                 */
+                float_X effectiveCharge = chargeState + float_X(1.0);
                 /* effective principal quantum number (unitless) */
-                float_X nEff = protonNumber / math::sqrt(float_X(2.0) * iEnergy );
+                float_X nEff = effectiveCharge / math::sqrt(float_X(2.0) * iEnergy );
```
Comment (n01r) [inline src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp]: `float_X const`

File: src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp [hunk 7]
```diff
@@ -79,21 +79,25 @@ namespace ionization
                 /* electric field in atomic units - only absolute value */
                 float_X eInAU = math::abs(eField) / ATOMIC_UNIT_EFIELD;
```
Comment (n01r) [inline src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp]: `float_X const`

File: docs/source/models/field_ionization_effective_potentials.py (hunk 1 above)
Comment (n01r) [inline docs/source/models/field_ionization_effective_potentials.py]: fixed

File: docs/source/models/field_ionization.rst (hunk 2 above)
Comment (n01r) [inline docs/source/models/field_ionization.rst]: fixed

File: docs/source/models/collisional_ionization.rst [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+Collisional Ionization
+======================
+
+LTE models
+----------
+
+.. moduleauthor:: Marco Garten
+
+Implemented LTE Model: Thomas-Fermi Ionization according to [More1985]_
```
Comment (PrometheusPi) [inline docs/source/models/collisional_ionization.rst]: OPTIONAL:
Please remove capitalization `model`, `ionization`

File: docs/source/models/collisional_ionization.rst [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+Get started here https://github.com/ComputationalRadiationPhysics/picongpu/wiki/Ionization-in-PIConGPU
+
+NLTE Models
```
Comment (PrometheusPi) [inline docs/source/models/collisional_ionization.rst]: OPTIONAL:
either write `model` or capitalize line 6

**Comment by n01r** [comment]
Thanks! The script refactoring is very helpful!

File: docs/source/models/field_ionization.rst [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+.. sectionauthor:: Marco Garten
+.. moduleauthor:: Marco Garten
+
+Get started here https://github.com/ComputationalRadiationPhysics/picongpu/wiki/Ionization-in-PIConGPU
+
+PIConGPU features an adaptable ionization framework for arbitrary and combinable ionization models.
+
+.. note::
+
+    Most of the calculations and formulae in this section of the docs are done in the system of **Atomic Units (AU)**.
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: OPTIONAL:
``**Atomiv Units (AU)** system``?

File: docs/source/models/field_ionization.rst [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+.. table:: **Atomic Unit System**
+    :widths: auto
+    :name: atomic_units
+
+    ================    =====
+     AU                 SI
+    ================    =====
+    length              :math:`5.292 \cdot 10^{-11}\,\mathrm{m}`
+    time                :math:`2.419 \cdot 10^{-17}\,\mathrm{s}`
+    energy              :math:`4.360 \cdot 10^{-19}\,\mathrm{J}\quad` (= 27.21 eV = 1 Rydberg)
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: REQUIRED:
The Hartree energy is 4.3 10^-18 J

Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: but time is  correct

File: docs/source/models/field_ionization.rst [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+Tunneling ionization describes the process during which an initially bound electron quantum-mechanically tunnels through a potential barrier of finite height.
+
+Keldysh
+"""""""
+
+.. math::
+
+    \Gamma_\mathrm{K} = \frac{\left(6 \pi\right)^{1/2}}{2^{5/4}} E_\mathrm{ip} \left( \frac{F}{(2 E_\mathrm{ip})^{3/2}} \right)^{1/2} \exp\left(-\frac{2 \left(2 E_\mathrm{ip}\right)^{3/2}}{3 F}\right)
+
+According to the equation (9) in [BauerMulser1999]_ the Keldysh ionization rate has been implemented. See also [Keldysh]_ for the original work.
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: OPTIONAL:
Better turn parts around:
The Keldysh ionization rate has been implemented according to the equation (9) in [BauerMulser1999]_.

File: docs/source/models/field_ionization.rst [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+
+.. math::
+
+    \begin{align}
+        \Gamma_\mathrm{ADK} &= \underbrace{\sqrt{\frac{3 n^{*3} F}{\pi Z^3}}}_\text{lin. pol.} \frac{F D^2}{8 \pi Z} \exp\left(-\frac{2Z}{3n^{*3}F}\right) \\
+        D &\equiv \left( \frac{4 \exp Z^3}{F n^{*4}} \right)^{n^*} \hspace{2cm} n^* = \frac{Z}{\sqrt{2 E_\mathrm{ip}}}
+    \end{align}
+
+We implemented equation (7) from [DeloneKrainov]_ which is a *simplified result assuming s-states* (since we have no atomic structure implemented, yet).
+Leaving out the pre-factor distinguishes ``ADKCircPol`` from ``ADKLinPol``.
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: OPTIONAL:
This statement is not clear to me, please specify what you mean with this sentence.

File: docs/source/models/field_ionization.rst [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+    .. attention::
+
+        Take care that :math:`Z` denotes the **residual ion charge** and not the proton number of the nucleus!
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: OPTIONAL:
better `Take care` -> `Be aware that `

File: docs/source/models/field_ionization.rst [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+
+In the following comparison one can see the ``ADKLinPol`` ionization rates for the transition from Carbon II to III (meaning 1+ to 2+).
+For a reference the rate for Hydrogen as well as the barrier suppression field strengths :math:`F_\mathrm{BSI}` have been plotted.
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: Required
either rate**s** or **has** been plotted

File: docs/source/models/field_ionization.rst [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+    :alt: Comparison of ADK ionization rates for Carbon-II and Hydrogen
+
+    Comparison of ADK ionization rates for Carbon-II and Hydrogen.
+
+When we account for orbital structure in shielding of the ion charge :math:`Z` according to [ClementiRaimondi1963]_ in ``BSIEffectiveZ`` the barrier suppression field strengths of Hydrogen and Carbon-II are very close to one another.
+One would expect much earlier ionization of Hydrogen due to lower ionization energy. The following image shows how this can be explained by the shape of the ion potential that is assumed in this model.
+
+.. figure:: field_ionization_effective_potentials.svg
+    :name: field_ionization_effective_potentials
+    :alt: Effective atomic potentials of Carbon-II and Hydrogen in homogenous electric field
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: REQUIRED:
not `homogenous` but `homogeneous`

File: docs/source/models/field_ionization.rst [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+
+    Effective atomic potentials of Carbon-II and Hydrogen in homogenous electric field :math:`F_\mathrm{BSI}` (C-II).
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization.rst]: REQUIRED:
same here

File: docs/source/models/field_ionization_comparison_c_ii_ionization.py [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+    # RTD default textwidth: 800px
+    'figure.figsize': [12, 8]
+}
+mpl.rcParams.update(params)
+
+
+def ADK_rate_simple(Z, E_i, F):
+    """
+    ADK tunneling rate from [DeloneKrainov1998] that has been simplified
+    for s-states.
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization_comparison_c_ii_ionization.py]: REQUIRED:
please add a comment on the parameters

File: docs/source/models/field_ionization_comparison_c_ii_ionization.py [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    On execution this file produces the image
+    `field_ionization_comparison_c_ii_ionization.svg`
+    for the PIConGPU documentation.
+    """
+
+    # Atomic units
+    E_AU = 27.2  # eV
+    F_AU = 5.1422e11  # V/m
+    I_AU = 3.5095e16  # W/cm^2
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization_comparison_c_ii_ionization.py]: QUESTION:
If this is in A.U. why do you give a SI unit?

File: docs/source/models/field_ionization_comparison_c_ii_ionization.py [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+               label="$F_\mathrm{BSI}$ C (eff)", linestyles="--")
+
+    plt.ylim([ymin, ymax])
+    plt.xlim([1e-2, 1e1])
+    plt.yscale("log")
+    plt.xscale("log")
+    plt.ylabel("ionization rate $\Gamma$ [Hz]")
+    plt.xlabel("field strength $F$ [AU = 5.1422$\cdot 10^{11}$ V/m]")
+    plt.legend(loc="best")
+
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization_comparison_c_ii_ionization.py]: OPTIONAL:
you could add  `plt.tight_layout()`

File: docs/source/models/field_ionization_effective_potentials.py [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+    # RTD default textwidth: 800px
+    'figure.figsize': [12, 8]
+}
+mpl.rcParams.update(params)
+
+
+def V_eff(x, Z_eff, F):
+    """
+    Effective radially symmetric nuclear potential under the influence of a
+    homogenous external electric field.
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization_effective_potentials.py]: REQUIRED:
typo homogen**e**ous

File: docs/source/models/field_ionization_effective_potentials.py [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+
+    Unit: 1 AU = 27.2 eV
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization_effective_potentials.py]: REQUIRED:
Please add brief comment on parameters

File: docs/source/models/field_ionization_effective_potentials.py [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+    plt.legend(loc="best")
+    plt.text(xmin+1, -E_H+.05, r"$E_\mathrm{i}$ H")
+    plt.text(xmin+1, -E_CII+.05, r"$E_\mathrm{i}$ C-II (Z_eff)")
+    plt.text(xmin+1, -2.1, r"$V_\mathrm{eff} = -\frac{Z}{|x|} + Fx$")
+    plt.text(xmin+1, -2.6, r"$\mathrm{\ with\ } F = F_\mathrm{BSI}$(C-II)")
+    plt.xlim([-10., 10.])
+    plt.ylim([-3., 1.])
+    plt.ylabel("$V_\mathrm{eff}$ [AU = Rydberg]")
+    plt.xlabel("$x$ [AU = Bohr radii]")
+
```
Comment (PrometheusPi) [inline docs/source/models/field_ionization_effective_potentials.py]: OPTIONAL:
add ``plt.tight_layout()``

File: docs/source/models/collisional_ionization.rst (hunk 9 above)
Comment (ax3l) [inline docs/source/models/collisional_ionization.rst]: this header is fine, line 6 needs capitalization

File: docs/source/models/collisional_ionization.rst [hunk 24]
```diff
@@ -0,0 +1,30 @@
+.. _model-collisionalIonization:
+
+Collisional Ionization
+======================
+
+LTE models
```
Comment (ax3l) [inline docs/source/models/collisional_ionization.rst]: Model needs capitalization

**Comment by ax3l** [comment]
@n01r pls ammend the changes (are all docs-only) to your last commit when finished (careful: check the branch out fresh first since I force-pushed)

File: docs/source/models/field_ionization.rst (hunk 11 above)
Comment (n01r) [inline docs/source/models/field_ionization.rst]: thanks for catching that typo!

File: docs/source/models/field_ionization.rst (hunk 16 above)
Comment (n01r) [inline docs/source/models/field_ionization.rst]: `homogenous` is actually an old-fashioned term for `homologous` from biology ^^;

thanks!

File: docs/source/models/field_ionization_comparison_c_ii_ionization.py (hunk 19 above)
Comment (n01r) [inline docs/source/models/field_ionization_comparison_c_ii_ionization.py]: that **is** the unit ^^

File: docs/source/models/field_ionization_comparison_c_ii_ionization.py (hunk 20 above)
Comment (n01r) [inline docs/source/models/field_ionization_comparison_c_ii_ionization.py]: offline: @ax3l says no

File: docs/source/models/field_ionization_effective_potentials.py (hunk 23 above)
Comment (n01r) [inline docs/source/models/field_ionization_effective_potentials.py]: as discussed offline: will not implement as it should rather be set manually

**Comment by n01r** [comment]
> check the branch out fresh first

Yep, did that immediately.

Thanks for the feedback! I hope it can be merged now :+1:

File: docs/source/models/field_ionization_effective_potentials.py (hunk 23 above)
Comment (PrometheusPi) [inline docs/source/models/field_ionization_effective_potentials.py]: Okay

File: docs/source/models/field_ionization_comparison_c_ii_ionization.py (hunk 20 above)
Comment (PrometheusPi) [inline docs/source/models/field_ionization_comparison_c_ii_ionization.py]: Okay

File: docs/source/models/field_ionization_comparison_c_ii_ionization.py (hunk 19 above)
Comment (PrometheusPi) [inline docs/source/models/field_ionization_comparison_c_ii_ionization.py]: Okay - I was confused by the relation but know it makes sense

**Comment by ax3l** [comment]
@PrometheusPi pls update your review :)

**Pull request merged**